backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js

```diff
@@ -0,0 +1,388 @@
+//
+//  index.test.js
+//  CompactConnect
+//
+//  Created by InspiringApps on 7/22/2024.
+//
+
+const {
+    expect,
+    testFilename,
+    runLambda,
+    lambdaConfig
+} = require('./config');
+
+// ================================================================================================
+// =                                        SETUP                                                 =
+// ================================================================================================
+const environments = {
+    csg: {
+        prod: {
+            webFrontend: `app.compactconnect.org`,
+            dataApi: `api.compactconnect.org`,
+            s3Upload: ``,
+        },
+        test: {
+            webFrontend: `app.test.compactconnect.org`,
+            dataApi: `api.test.compactconnect.org`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-ivslg22f2yfz.s3.amazonaws.com`,
+        },
+    },
+    ia: {
+        prod: {
+            webFrontend: `app.jcc.iaapi.io`,
+            dataApi: `api.jcc.iaapi.io`,
+            s3Upload: ``,
+        },
+        test: {
+            webFrontend: `app.test.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+        justin: {
+            webFrontend: `app.justin.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+    },
+};
+const buildCspHeaders = (environment) => {
+    const dataApiUrl = (environment?.dataApi) ? `https://${environment.dataApi}` : '';
+    const s3Url = (environment?.s3Upload) ? `https://${environment.s3Upload}` : '';
+    const cspDefaultSrc = [
+        '\'none\'',
+    ].join(' ');
+    const cspManifestSrc = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrc = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrcElem = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrcAttr = [
+        '\'self\'',
+    ].join(' ');
+    const cspWorkerSrc = [
+        '\'self\'',
+    ].join(' ');
```

Comment:
I assume your goal here is to let this be extendable going forward? It's definitely less readable to me than just e.g., `cspDefaultSrc = "'none'";` but perhaps worth it. Mind adding a comment at the top just indicating why it's written like this?

Reply:
Added a comment in the code.